GoogleForCreators/web-stories-wp
# Fix entering edit mode via click for Safari

## Summary
Entering edit mode for Safari was using `Document.caretRangeFromPoint()` that seems to be deprecated (according to the `lib.dom.d.ts` file my IDE is using) or at least not recommended for use in production due to potential issues. It is currently failing to work correctly in our editor and causing unexpected behaviour.

This PR stops using this method and enters the edit mode by setting the cursor to index `0` for Safari (and any browser that does not support `caretPositionFromPoint`).


## Relevant Technical Choices



## To-do

- [x] Investigate if there's an alternative for setting the cursor where the user clicked. -> @merapi found an alternative



## User-facing changes



## Testing Instructions
In Safari:
- Add a Text.
- Enter edit mode
- Verify you can select part of the text and mark it bold (or italic, etc.)


### QA






This PR can be tested by following these steps:

1. See the instructions above.

### UAT







This PR can be tested by following these steps:

1.

## Checklist




---



Fixes #7745
